diff --git a/src/store/actions.ts b/src/store/actions.ts
index 05ddc62d..49bd68f2 100644
--- a/src/store/actions.ts
+++ b/src/store/actions.ts
@@ -45,7 +45,7 @@ export const actions: ActionTree<RootState, RootState> = {
                 Array.isArray(payload.instances) &&
                 payload.instances.length
             ) {
-                commit('setConfigInstances', payload.instances)
+                commit('setConfigInstances', payload.instances.map((i) => ({ ...i, hostname: i.hostname ?? window.location.hostname, port: i.port ?? (window.location.port ? Number(window.location.port) : (window.location.protocol === 'https:' ? 443 : 80)) })))
             }
 
             return
diff --git a/src/store/types.ts b/src/store/types.ts
index 11cdd6d2..a90213f8 100644
--- a/src/store/types.ts
+++ b/src/store/types.ts
@@ -34,8 +34,8 @@ export interface ConfigJson {
 }
 
 export interface ConfigJsonInstance {
-    hostname: string
-    port?: number
+    hostname: string | null
+    port?: number | null
     path?: string
 }
 
